### ISSUE TITLE
Query Organization

### ISSUE THREAD

**Issue by ronnyek** [ChilliCream/hotchocolate#277]
**Is your feature request related to a problem? Please describe.**
I was just mainly curious if there was roughly any way to do what the graphql-dotnet project does for query organization. That is basically provide ObjectType for grouping query methods down into different levels. (https://graphql-dotnet.github.io/docs/getting-started/query-organization)

**Comment by michaelstaib** [comment]
Hi @ronnyek,

there are quite a view ways to do that. So, with the current release Version you can refer to fields in external types.

```csharp
public class PersonType : ObjectType<Person>
{
    protected override void Configure(IObjectDescriptor<Person> desc)
    {
        desc.Field<PersonXyz>(t => t.GetFriends(default, default));
    }
}
```

So, in the above example the `PersonType` exposes all properties and methods of `Person` and also Includes the `GetFriends` method from PersonXyz. You can do this with any type not only `Query` or `Mutation`. This lets you also share resolver logic, since if you write GetFriends generic enough it could server as a resolver for many types (we have an example for this if you want to see that in action).

With the upcoming Version 0.5.0 (should arrive end of this week) we are introducing a lot of new features for schema stitching. So, then you can do the following:

```csharp
public class PersonType : ObjectType<Person>
{
    protected override void Configure(IObjectDescriptor<Person> desc)
    {
        desc.Include<SecondType>();
       desc.Include<ThirdType>();
    }
}
```

This combined with the new executable directives lets you combine external rest services, external GraphQL services etc with your own GraphQL Schema to represent a single Schema. 

Basically you than could integrate an external service like the following:

```GraphQL
type Query {
   externalCall: Person @fetch(url: "http://fooservice/getPersons") @json(path: "persons.first()")
}

type Person {
 ....
}
```

So, by providing GraphQL SDL snippets like the following to our schema you can merge multiple schemas into one and integrate an external data provider without any code complex code.

Hope this clarifies what you can do now and what we are introducing this week.

**Comment by michaelstaib** [comment]
>  That is basically provide ObjectType for grouping query methods down into different levels.

Sorry misread your question. So, the solution that I stated is for breaking up .net code into different types and merging different schemas and types together. 

You can group your api like you want. There is no limit. 
You could in the most simple case create a non-generic query type like the following: 

```csharp
public class QueryType : ObjectType
{    
    protected override void Configure(IObjectDescriptor<Person> desc)
    {
        desc.Field("group").Type<GroupType>().Resolver(() => new object());
    }
}
```

But this should be possible with any GraphQL implementation. 

Did this help you?

**Comment by ronnyek** [comment]
Wow, awesome information. I could see how the configure like this could effectively enable the type/api inference.

I appreciate the feedback.

**Comment by ronnyek** [comment]
It looks like include may touch on this functionality, but the specific bits of the organization I was hoping to find were things that would let me do something like this
User->Accounts->AllActiveUsers()

And I also don't know if that even makes sense from a graphql perspective.

**Comment by michaelstaib** [comment]
I will create some demo code an will put it on GitHub.

**Comment by ronnyek** [comment]
That would be awesome. I love the progress you've made on this project. Makes me want to contribute in some way, shape, or form.

**Comment by ronnyek** [comment]
I think this makes sense:
https://github.com/ChilliCream/hotchocolate/issues/277#issuecomment-426268163

I did have a followup question about this... regarding basically making conventions for more easily generating api methods. In just tinkering, I built a reflection based convention and had GroupType hard coded, and its generating appropriate objects at runtime, but is there a way to generate the wrapper type (ObjectType<Whtaever>) for being able to specify to the .Type() method.

I was thinking about being able to build an object that I would markup like such: (using System.ComponentModel.Annotations where possible)
``` cs
[Display("Accounts")]
[GraphQLRoute("User/Accounts")]
public class GraphQLAccountsApi
{
      [GraphQLMethod("AllAccounts")]
      public IReadOnlyList<Account> AllAccounts() => ...
}
```

Seems like doing this with what you specified above would actually work, except I don't know how you'd actually be able define inferred `Account` type, and specify it when using descriptor for GraphQLAccountsApi to add fields etc.

**Comment by michaelstaib** [comment]
> Makes me want to contribute in some way, shape, or form.

Sure, there is lots to do and contributions in any form are welcome (Code, Documentation, Feature Requests etc....).

**Comment by ronnyek** [comment]
awesome... I'm all for helping out however I can... I left comments on a few potential ways we can achieve it, but ultimately this is your library, and am not trying to step on toes or anything =)

**Comment by LuvForAirplanes** [comment]
@michaelstaib What is the current way to organize GraphQL endpoints? [This example](https://graphql-dotnet.github.io/docs/getting-started/query-organization/) doesn't work in current versions, and I can't seem to make anything work. 😏
